datadog_checks_base/tests/base/checks/test_agent_check.py

```diff
@@ -50,6 +51,46 @@ def test_load_config():
         AgentCheck.load_config(':')
 
 
+def test_parse_special_values_in_load_config():
```

Comment:
Sorry for the noise, but, can we parameterize this test? With so many asserts if something breaks it will be unclear to know what exactly breaks unless we run multiple times until we catch them all. We can parameterize it with the `yaml_str`, `expected_object` and `instance_type` for the isinstance assertion.

Reply:
Thanks Enrico! I am solving this thread as well waiting for the updates we talked about offline 🎉